### PR TITLE
External SVG Icons

### PR DESCRIPTION
**Pull request by adiraomj**
Implemented enhancement #2 

Added options for SVG icons, implemented it in templates & updated styles.

Included the [svgxuse](https://github.com/Keyamoon/svgxuse) for browser compatibility.

### CODE REVIEW

File: footer.php [hunk 1]
```diff
@@ -47,9 +47,17 @@
 </div><!-- #page -->
 
 <?php if( siteorigin_setting('navigation_scroll_to_top') ) : ?>
-	<div id="scroll-to-top">
-		<img src="<?php echo get_template_directory_uri() ?>/images/up-arrow.svg" width="24px" height="24px" />
-	</div>
+	<?php if( siteorigin_setting('navigation_scroll_to_top_icon') ) : ?>
+		<div id="scroll-to-top">
+			<svg>
+				<use xlink:href="<?php echo siteorigin_setting( 'navigation_scroll_to_top_icon' ); ?>">
```
Comment (gregpriday) [inline footer.php]: Because this is user entered data, it should be escaped with `esc_url`.

File: inc/template-tags.php [hunk 2]
```diff
@@ -311,32 +311,40 @@ function siteorigin_north_category_transient_flusher() {
...
-			?>
-			<svg version="1.1" class="svg-icon-menu" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 24 24" style="enable-background:new 0 0 24 24;" xml:space="preserve">
-				<path class="line-1" d="M3,5h18c0.3,0,0.5,0.1,0.7,0.3C21.9,5.5,22,5.7,22,6s-0.1,0.5-0.3,0.7C21.5,6.9,21.3,7,21,7H3 C2.7,7,2.5,6.9,2.3,6.7C2.1,6.5,2,6.3,2,6s0.1-0.5,0.3-0.7C2.5,5.1,2.7,5,3,5z"/>
-				<path class="line-2" d="M3,11h18c0.3,0,0.5,0.1,0.7,0.3S22,11.7,22,12s-0.1,0.5-0.3,0.7S21.3,13,21,13H3c-0.3,0-0.5-0.1-0.7-0.3 C2.1,12.5,2,12.3,2,12s0.1-0.5,0.3-0.7C2.5,11.1,2.7,11,3,11z"/>
-				<path class="line-3" d="M3,17h18c0.3,0,0.5,0.1,0.7,0.3S22,17.7,22,18s-0.1,0.5-0.3,0.7S21.3,19,21,19H3c-0.3,0-0.5-0.1-0.7-0.3 C2.1,18.5,2,18.3,2,18s0.1-0.5,0.3-0.7C2.5,17.1,2.7,17,3,17z"/>
-			</svg>
-			<?php
+			if ( siteorigin_setting( 'responsive_menu_icon' ) ): ?>
+				<svg class="svg-icon-menu">
+					<use xlink:href="<?php echo siteorigin_setting( 'responsive_menu_icon' ); ?>">
```
Comment (gregpriday) [inline inc/template-tags.php]: Same here, this needs an `esc_url`, or some other type of escaping if the user should be permitted to enter non URL values.

**Comment by gregpriday** [comment]
I'll test this out soon, but for now there are those 2 escaping issues you can take a look at.

**Comment by gregpriday** [comment]
So here are my suggestions for this PR after using it for a bit.

First, I've changed a few things with the settings framework and added a lot more settings, so you should pull the latest develop into this branch. They're all in the develop branch now.

We should have all the icon settings inside a new theme settings section called `icons`. I think you even mentioned this in the past. This would also make it much easier for us to add color options for the icons without bulking up the other sections too badly. You could even go ahead and add these color options if you're familiar with how the whole SASS/Settings system works.

Another thing to investigate would be using a media upload field instead of a text field for each icon. I think there might be some issues with this because the default media field doesn't work with SVG, but its worth investigating. Maybe we can add some sort of file setting to the SiteOrigin Settings library, just let me know if you need me to do that.

Lastly, as a change to my code, I think we should move the icon output from the footer into the `siteorigin_north_display_icon` function. I did this in a branch, so you're more than welcome to cherry pick my change.

https://github.com/siteorigin/siteorigin-north/commit/a7e7d4596e65c05d23a8fdecc98ad7e30687c505

**Comment by adiraomj** [comment]
I'll look into everything you have mentioned.

With reference to 5795854c3b9b55e1cd713b901633b9f537b93f24.
From what I understand, users will be setting the SVG icons color before uploading them. So if we set the color in the stylesheet, won't we be overwriting it. Let me know if I'm mistaken.

Also, adding all the icons to `siteorigin_north_display_icon` was a great idea.

**Comment by gregpriday** [comment]
So ideally we'd want to have the colour value blank, in which case it would take on either the default value of the stylesheet or the default colour of the icon (we should probably try ensure it takes on the default colour of the icon).

If the user sets a colour in our theme settings, then the icon should take on that colour. That's ideal for users who might have downloaded the icon from some sort of icon pack and don't know how to edit the files to change the colour.

**Comment by adiraomj** [comment]
To build the icon settings, I was thinking we can have a file `upload field` to upload a spritesheet.

And then use text fields for users to input the icon's identifier eg: `icon-menu`. 

The `use` element needs the identifier to function. We can also write up a small tutorial on how to build & download spritesheets using the icomoon app.

Also, where can I find a list of field types available to me with the settings framework.

**Comment by adiraomj** [comment]
I have made changes to the PR to allow users to upload images as well as svg icons.

svg is supported via the [SVG Support](https://wordpress.org/support/plugin/svg-support) plugin.

made changes to the close search icon, as it was not functioning properly. Made minor changes to the CSS and JS associated with it.

**Comment by adiraomj** [comment]
Fixed all merge conflicts

**Comment by Misplon** [comment]
@adiraomj This is looking great, thanks. Could you have another look at conflicts for us, please? We'll merge once they're resolved.

**Comment by adiraomj** [comment]
@Misplon I have fixed all merge conflicts. Please test it out and let me know if you run into any problems.

**Comment by Misplon** [comment]
Sorry @adiraomj, one more time, please.

**Pull request merged**